content/techniques/file-upload.md

```diff
@@ -41,6 +41,113 @@ The `FileInterceptor()` decorator takes two arguments:
 
 > warning **Warning** `FileInterceptor()` may not be compatible with third party cloud providers like Google Firebase or others.
 
+#### File Validation
+
+Often times it can be useful to validate incoming file metadata, like file size or file mimetype. You can create your own [`Pipes`](https://docs.nestjs.com/pipes) and add them to the `FileInterceptor` decorator to achieve this, considering the `value` inside each pipe will be the file object. The example below demonstrates how a basic file size validator pipe could be implemented:
```

Comment:
```suggestion
Often times it can be useful to validate incoming file metadata, like file size or file mime-type. For this, you can create your own [Pipe](https://docs.nestjs.com/pipes) and bind it to the parameter annotated with the `UploadedFile` decorator. The example below demonstrates how a basic file size validator pipe could be implemented:
```